### DISCUSSION TITLE
(error) ERR Invalid password

### DISCUSSION THREAD

**Discussion by epubreader** [apache/kvrocks#2462] [General]
docker service create \
  --name kvrocks \
  --publish 6666:6666 \
  --mount type=bind,source=/home/owner/docker/kvrocks/data,target=/data  \
  --constraint 'node.labels.name==mongodb2' \
  --mode global \
  --restart-condition on-failure \
  --with-registry-auth \
  --env KVROCKS_MASTERAUTH=drycc \
  --env KVROCKS_REQUIREPASS=drycc \
  --env KVROCKS_DIR=/data \
  apache/kvrocks:2.9.0 --requirepass $KVROCKS_REQUIREPASS --masterauth $KVROCKS_MASTERAUTH

  Run the auth command, then
  (error) ERR Invalid password

**Comment by git-hulk** [comment]
@epubreader You need to rewrite the entry point, can refer to: https://github.com/apache/kvrocks-controller/blob/unstable/scripts/docker/docker-compose.yml#L27

**Reply by PragmaTwice** [reply]
hmm I cannot get your point.

**Reply by git-hulk** [reply]
My point is that need to rewrite the entry point of service if want to use environments instead of using environments directly in the command line.

**Reply by PragmaTwice** [reply]
ahh I see.

**Comment by PragmaTwice** [comment]
The environment variable you set is inside the container, but here `$KVROCKS_REQUIREPASS` will be resolved in your host environment.

So it's due to your wrong use of shell.

please try this: `docker run -it --rm -p 6666:6666 apache/kvrocks:2.9.0 --requirepass testpw`

or you can add single quote mark to your `$XXX` to prevent resolving in host env.

**Reply by epubreader** [reply]
This may leak the password. I changed to mount the configuration file, but it still failed.   (error) ERR Invalid password

  docker service create \
  --name kvrocks \
  --publish 6666:6666 \
  --mount type=bind,source=/home/owner/docker/kvrocks/,target=/var/lib/kvrocks  \
  --mode global \
  --restart-condition on-failure \
  --with-registry-auth \
  apache/kvrocks:2.9.0

**Reply by PragmaTwice** [reply]
Again, it's not related to kvrocks.

Please refer to the documentation of docker and shell.

**Comment by epubreader** [comment]
> For persisted data, prefer using a docker volume over a bind-mounted volume to avoid file permission or performance problems. 
why bind-mounted volume will affect performance problems?
